docs/proposals/networking/multiclusterservice.md

```diff
@@ -0,0 +1,251 @@
+
+const (
+	// ExposureTypeCrossCluster means a service will be accessible across clusters.
+	ExposureTypeCrossCluster ExposureType = "CrossCluster"
+
+	// ExposureTypeLoadBalancer means a service will be exposed via an external
+	// load balancer.
+	ExposureTypeLoadBalancer ExposureType = "LoadBalancer"
+)
+
+// ExposurePort describes which port will be exposed.
+type ExposurePort struct {
+	// Name is the name of the port that needs to be exposed within the service.
+	// The port name must be the same as that defined in the service.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// Port specifies the exposed service port.
+	// +optional
+	Port int32 `json:"port,omitempty"`
+}
+
+// ExposureRange describes a list of clusters where the service is exposed.
+// Now supports selecting cluster by name, leave the room for extend more methods
+// such as using label selector.
+type ExposureRange struct {
+	// ClusterNames is the list of clusters to be selected.
+	// +optional
+	ClusterNames []string `json:"clusterNames,omitempty"`
+}
+```
+
+### Shared interfaces for MultiClusterService
+
+```go
+// MCSLoadBalancer is an abstract, pluggable interface for MultiClusterService load balancers.
+type MCSLoadBalancer interface {
+	// GetMCSLoadBalancer return whether the specified load balancer exists, and if so, what its status is.
+	GetMCSLoadBalancer(ctx context.Context, mcs *providernetworkingv1alpha1.MultiClusterService) (status *corev1.LoadBalancerStatus, exist bool, err error)
+	// EnsureMCSLoadBalancer creates a new load balancer.
+	EnsureMCSLoadBalancer(ctx context.Context, mcs *providernetworkingv1alpha1.MultiClusterService) (status *corev1.LoadBalancerStatus, err error)
+	// UpdateMCSLoadBalancer updates the specified load balancer.
+	UpdateMCSLoadBalancer(ctx context.Context, mcs *providernetworkingv1alpha1.MultiClusterService) (status *corev1.LoadBalancerStatus, err error)
+	// EnsureMCSLoadBalancerDeleted deletes the specified load balancer if it exists.
+	EnsureMCSLoadBalancerDeleted(ctx context.Context, mcs *providernetworkingv1alpha1.MultiClusterService) error
+}
+```
+
+### Service exposure
+
+1. Create Deployments and Services and propagate them to member clusters.
+2. Create the multiClusterService resource.
+3. Register the pod IP address with the LB instance using multicluster-cloud-provider-xxx.
+4. The user accesses the LB instance through Layer 4.
+
+```yaml
+apiVersion: networking.karmada.io/v1alpha1
+kind: MultiClusterService
+metadata:
+  name: foo
+spec:
+   ports:
+   - port: 80
+   types:
+   - LoadBalance
+```
+
+![image](statics/service-exposure.png)
+
+### Cross-clusters service discovery
+
+1. Create Deployments and Services and propagate them to member clusters.
+2. Create the multiClusterService resource.
+3. The controller list/watch to the multiClusterService resource, generates ServiceExport and propagate it to the source clusters, and generates ServiceImport and propagate it to the destination clusters.
```

Comment:
> , generates ServiceExport and propagate it to the source clusters, and generates ServiceImport and propagate it to the destination clusters.

How does this new controller work in conjunction with existing` MCS controllers`?

Reply:
Add more info, can you help review it again?